packages/react-hooks/src/useTarget.ts

```diff
@@ -0,0 +1,58 @@
+import { Point, getDocument, getWindow } from '@uifabric/utilities';
+import * as React from 'react';
+
+export type Target = Element | string | MouseEvent | Point | null | React.RefObject<Element>;
+
+/**
+ * Hook to calculate and cache the target element specified by the given target attribute,
+ * as well as the target element's (or host element's) parent window
+ * @param target- Target selector passed to the component as a property, describing the element that
+ * the callout should target
+ * @param hostElement- The callout's host element, used for determining the parent window.
+ */
+export function useTarget<TElement extends HTMLElement = HTMLElement>(
+  target: Target | undefined,
+  hostElement?: React.RefObject<TElement | null>,
+): Readonly<[React.RefObject<Element | MouseEvent | Point | null>, React.RefObject<Window | undefined>]> {
+  const previousTargetProp = React.useRef<
+    Element | string | MouseEvent | Point | React.RefObject<Element> | null | undefined
+  >();
+
+  const targetRef = React.useRef<Element | MouseEvent | Point | null>(null);
+  /**
+   * Stores an instance of Window, used to check
+   * for server side rendering and if focus was lost.
+   */
+  const targetWindowRef = React.useRef<Window>();
```

Comment:
Should we switch this to use the new `useWindow` hook from `react-window-provider`? I think it was added after this PR started.